app/app_helper.rb

```diff
@@ -0,0 +1,22 @@
+def current_user
+  @current_user ||= User.get(session[:user_id])
+end
+
+def find_user(username_or_email)
+  user = User.first(username: username_or_email)
+  user = User.first(email: username_or_email) if user.nil?
+  user
+end
+
+def sign_up_error_message(username, email, password, password_confirmation)
+  flash.now[:alert] = if User.all(username: username).any?
```

Comment:
I feel like this logic is in the wrong place. You're attempting to create a user, and if the user isn't successfully created, you're analysing the inputs to suggest what might have gone wrong.

A more normal model is that you validate the inputs before attempting to create the user, and creating the user fails, you analyse the exception to extract a suitable error message, even if that message is just "Unexpected error"